main.cpp

```diff
@@ -131,233 +169,381 @@ struct CarWash
  This usually means you have to use identical types for all variables used in an expression that is producing that conversion warning.
  */
 
+// 
+// UDT No. 1 
+// 
+
+struct VolumeKnob
+{
+    int amountTurned;
+
+    int turnRight(bool)
+    {
+        return amountTurned;
+    }
+};
 
 struct Keyboard 
 {
-    //5 properties:
-    //    - midi device (std::string)
     std::string midiDevice = "akai";
-    //    - volume (double)
-    double volume = 9.797;
-    //    - amount of modulation (float)
+    int volume = 9;
     float amtModulation = 2.f;
-    //    - number of keys (int)
     int numKeys = 88;
-    //    - sustain (double)
     double amtSustain = 4.66663;
-    //3 things it can do:
-    //    - adjust volume   // Difference in initial volume setting and volume end value
-    double adjustVolume(double volumeKnobDiff);
-    //    - pitch shift    // Shirt from initial pitch based on note played with no modulation
+    
+    void adjustVolume(int volumeKnobDiff);
     float pitchShift(float intendedPitch = 0);
-    //    - sustain
     void sustian();
+
+    VolumeKnob volumeKnob;
+};
+
+void Keyboard::adjustVolume(int volumeKnobDiff) 
+{
+    if(volumeKnob.turnRight(true) > 0)
+    {
+        volume += volumeKnobDiff;
+    }
+    else if(volumeKnob.turnRight(false) < 0)
+    {
+        volume -= volumeKnobDiff;
+    }
+}
+
+// 
+// UDT No. 2
+// 
+
+struct TremoloBar
+{
+    void lift(){}
+    void depress(){}
 };
 
 struct ElectricGuitar 
 {
-    //5 properties:
-    //    - amount of pickups (int) 
     int amtPickups = 2;
-    //    - tremolo bar position (int)
-    int tremoloBarPosition = -2;
-    //    - string vibration (double)
+    float tremoloBarPosition = 0.f;
     double stringVibration = 82.947;
-    //    - tone knob (int)
     int toneKnobPosition = 0;
-    //    - pickup selector (int)
     int pickupSelection = 3;
-    //3 things it can do:
-    //    - capture string vibrations
+    
     void captureStringVibrations();
-    //    - dial in/out treble frequencies    // Select treble frequency level
     int dialInOutTrebleFreq(int initToneValue);
-    //    - adjust string tension    // Returns difference of string tension before and after use of tremolo bar
-    float adjustStringTension(float initStringTension);
+    float adjustStringTension(bool, float tremoloBarPositionDifference);
+
+    TremoloBar tremoloBar;
+};
+
+float ElectricGuitar::adjustStringTension(bool tensionIncreased, float tremoloBarPositionDifference)
+{
+    if(tensionIncreased == true)
+    {
+        tremoloBar.lift();
+    }
+    else
+    {
+        tremoloBar.depress();
+    }
+    
+    return tremoloBarPosition += tremoloBarPositionDifference;
+}
+
+// 
+// UDT No. 3
+// 
+
+struct washerDryerDoor
+{
+    void hotAirGasketEngage(){}
+    void waterGasketEngage(){}
```

Comment:
Verbs are the first word in a sentence describing an action.
`engageHotAirGasket`
`engageWaterGasket`
- [ ] revise